src/core/server/saved_objects/service/lib/update_objects_spaces.ts

```diff
@@ -260,14 +292,32 @@ export async function updateObjectsSpaces({
     ? await client.bulk({ refresh, body: bulkOperationParams, require_alias: true })
     : undefined;
 
+  // Delete aliases if necessary, ensuring we don't have too many concurrent operations running.
+  const mapper = async ({ type, id, namespaces, deleteBehavior }: ObjectToDeleteAliasesFor) =>
+    deleteLegacyUrlAliases({
+      mappings,
+      registry,
+      client,
+      getIndexForType,
+      type,
+      id,
+      namespaces,
+      deleteBehavior,
```

Comment:
That's a shame we can't `bulkDeleteLegacyUrlAliases`, but given that `namespaces` and `deleteBehavior` can be different per-object, I'm not even sure this would be doable with a single query/script.